overlord/snapstate/component.go

```diff
@@ -58,15 +58,17 @@ func InstallComponentPath(st *state.State, csi *snap.ComponentSideInfo, info *sn
 		return nil, err
 	}
 
+	kmodComps := snapst.Sequence.ComponentsWithTypeForRev(snapst.Current, snap.KernelModulesComponent)
 	snapsup := SnapSetup{
-		Base:        info.Base,
-		SideInfo:    &info.SideInfo,
-		Channel:     info.Channel,
-		Flags:       flags.ForSnapSetup(),
-		Type:        info.Type(),
-		Version:     info.Version,
-		PlugsOnly:   len(info.Slots) == 0,
-		InstanceKey: info.InstanceKey,
+		Base:                            info.Base,
+		SideInfo:                        &info.SideInfo,
+		Channel:                         info.Channel,
+		Flags:                           flags.ForSnapSetup(),
+		Type:                            info.Type(),
+		Version:                         info.Version,
+		PlugsOnly:                       len(info.Slots) == 0,
+		InstanceKey:                     info.InstanceKey,
+		PreUpdateKernelModuleComponents: kmodComps,
```

Comment:
my preference is still slightly to set from a relevant task(s) if possible instead of from every single place